src/ArticleQualityService.ts

```diff
@@ -65,7 +65,7 @@ class ArticleQualityService {
     const batchScores = await asyncPool(
       this.maxWorkers,
       batchRevisions,
-      input => this.getOresScores(input)
+      input => this.provideScores(input)
     );
 
     // Assign the scores to the revisions
```

Comment:
Basically, all the lines from here and until the return could be improved and rewritten (and I think perhaps we should open a task for it).

But, more specifically, we can make some initially small changes to the loop inside the mapping (`.map`, should be replaced by `.forEach`) that goes over `batchScores` (should probably be renamed to `batchResponses` to fit the type name) to make it work with an array of `OresScoresResponse` (should also be renamed, see next comments) rather than a single object.

I cannot comment on lines that were not changed in this PR, but you can wrap the `for` loop that iterates over the `Object.entries` of `scores`, with another one that iterates over the  `responses`. While we are at it, let's also rename the unimaginative `key` and `value` to something more descriptive like `revId` and `content`. This should all probably look something like:

**Before:**
```ts
batchScores.map(scores => {
  for (const [key, value] of Object.entries(scores || {})) {
    //...
  }
});
```

**After:**
```ts
batchResponses.forEach(responses => {
  for (const response of responses) {
     for (const [revId, content] of Object.entries(response || {})) {
       //...
     }
  }  
});
```

Of course, this end result is still not ideal (3 levels of nesting, mutating a variable out of the scope of the loop/function), and I would be super happy if you could go the extra mile and move this away from the imperative style (iteratively).

Reply:
Done. It required some explicit typecasting since typescript wasn't happy iterating over `responses` without it